src/main/java/spoon/support/compiler/jdt/TreeBuilderCompiler.java

```diff
@@ -72,6 +72,8 @@
 			requestor.acceptResult(unit.compilationResult);
 		}
 
-		return this.unitsToProcess;
+		CompilationUnitDeclaration[] unitsToReturn = new CompilationUnitDeclaration[this.totalUnits];
+		System.arraycopy(this.unitsToProcess, 0, unitsToReturn, 0, this.totalUnits);
```

Comment:
I checked the content of the `unitsToProcess` array and it always has non-null elements in index 0 to `this.totalUnits`.  The remainder of the elements in the array are `null`.  I can change it to scan the array and  only copy non-null values, but it won't be as efficient as the `System.arraycopy`.